lib/ansible/plugins/inventory/openstack.py

```diff
@@ -186,10 +194,20 @@ def parse(self, inventory, loader, path, cache=True):
                 cloud_inventory.clouds = new_clouds
 
             expand_hostvars = self._config_data.get('expand_hostvars', False)
-            fail_on_errors = self._config_data.get('fail_on_errors', False)
+            exit_on_errors = self._config_data.get('exit_on_errors', False)
+            fail_on_errors = exit_on_errors or \
+                    self._config_data.get('fail_on_errors', False)
 
-            source_data = cloud_inventory.list_hosts(
-                expand=expand_hostvars, fail_on_cloud_config=fail_on_errors)
+            try:
+                source_data = cloud_inventory.list_hosts(
+                    expand=expand_hostvars, fail_on_cloud_config=fail_on_errors)
+            except:
```

Comment:
could do ``except Exception as e``, add ``e.msg`` to the sys.exit message and then ``raise e`` in the else ?